packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -672,6 +718,259 @@ export const addMetadata = (payload: MetadataAddPayload) => (dispatch: Dispatch)
     return dispatch(addAccountMetadata(payload));
 };
 
+type EnabledDeviceMetadata = Extract<DeviceMetadata, { status: 'enabled' }>;
+type MissingDeviceFile = { type: 'device'; state: string } & Pick<
+    EnabledDeviceMetadata,
+    'aesKey' | 'fileName'
+>;
+type MissingAccountFile = { type: 'account' } & Pick<
+    AccountMetadata,
+    'aesKey' | 'fileName' | 'key'
+>;
+type MissingFileInfo = MissingDeviceFile | MissingAccountFile;
+
+// checks if every labelable entity has a file in storage already
+const getMissingFiles =
+    (filesList: string[] | undefined) => (dispatch: Dispatch, getState: GetState) => {
+        const provider = dispatch(getProvider());
+        if (!provider) {
+            return [];
+        }
+
+        const { device } = getState().suite;
+        if (!device || !device.state || !device.connected) {
+            return [];
+        }
+
+        const accounts = getState().wallet.accounts.filter(
+            a => a.deviceState === device.state && a.metadata.fileName,
```

Comment:
I wonder why is the `&& a.metadata.fileName` part needed here. I understand that you don't want to work sith accounts that don't have fileName, but is this the proper place to check it? Just academic disscussion

Reply:
Where would you check it?